modules/core/src/webgl/program.js

```diff
@@ -236,15 +260,8 @@ export default class Program extends Resource {
           const texture = uniform;
           const {textureIndex} = uniformSetter;
 
-          // TODO - this should be separated out from uniform setting, since it needs to be done
-          // before every draw even if uniforms have not changed
           texture.bind(textureIndex);
```

Comment:
Shouldn't we also remove `texture.bind(textureIndex);` , this is already done during `_bindTextures()` ?

Reply:
I tried it, looks like WebGL still needs *some* texture bound to the index to be able to set the uniform.

Reply:
Oh, that probably because `_bindTextures()` called afterwards, texture.bind() is not expensive so should be fine. Otherwise, we need 3 steps, extract changed uniforms, bind all textures, set changed uniforms.